matchers/matchers.go

```diff
@@ -147,3 +161,10 @@ func SupportedFileExtensions() []string {
 
 	return exts
 }
+
+func prepareTodosForRegex(todos []string) string {
+	if len(todos) == 0 {
+		panic("Empty list of todo strings")
+	}
+	return fmt.Sprintf("(?:%s)", strings.Join(todos, "|"))
+}
```

Comment:
This is not the place for such a function.

The problem is that you are writing it with the knowledge that all underlying todo matchers operate with regex.
This is an implementation detail and if at some point, we decide to migrate from regex to something else, this function here will cause a lot of issues.

Instead, this function can be defined in the `common` package as `arrayAsRegexAnyMatchExpression`.
The function declaration should instead be `[]string` and the specifics of turning the array of strings into a regex expression can be done for every todo matcher.

Reply:
Renamed and moved function to `common`